Factory reset crashes app when running from git worktree

## Bug Description

Factory reset operation causes app crash when running from a git worktree, making it impossible to reset workspace configuration during development.

## Steps to Reproduce

1. Run app from git worktree (e.g., `.loom/worktrees/extra-claude`)
2. Open the app and load a workspace
3. Trigger factory reset via menu: **File → Factory Reset Workspace**
4. App crashes with Rust panic

## Error Messages

**Console Log:**
```
[2025-10-15T22:12:28.941Z] [ERROR] Failed to reset workspace: Defaults directory not found: tried defaults and bundled resources
```

**Rust Panic:**
```
thread '<unnamed>' panicked at src/helpers/app_paths.rs:63:27:
failed to read cwd: Os { code: 2, kind: NotFound, message: "No such file or directory" }
```

## Root Cause

1. App is running from worktree: `.loom/worktrees/extra-claude`
2. Factory reset code looks for `defaults/` directory relative to current directory
3. Directory doesn't exist in worktree (only in main workspace root)
4. Error handling fails to prevent subsequent file operations
5. Current working directory becomes invalid, causing panic in `app_paths.rs`

## Expected Behavior

Factory reset should either:
- Find the correct `defaults/` directory in main workspace (preferred)
- Show user-friendly error message without crashing
- Gracefully degrade to allow user to continue using the app

## Impact

- **Severity**: High - crashes entire app
- **Affects**: Development workflow using git worktrees
- **Workaround**: Run app from main workspace directory instead of worktree

## Proposed Solutions

1. **Option 1**: Resolve `defaults/` path relative to app executable location
2. **Option 2**: Bundle defaults in Tauri resources (already attempted based on error)
3. **Option 3**: Improve error handling to prevent panic when defaults not found
4. **Option 4**: Search parent directories for `defaults/` folder

## Additional Context

Discovered during testing of Issue #125 (health monitoring). The health monitoring system itself works correctly - this is a separate issue with factory reset path resolution.

--- a/src-tauri/src/main.rs
+++ b/src-tauri/src/main.rs
@@ -312,15 +312,43 @@ fn copy_dir_recursive(src: &Path, dst: &Path) -> io::Result<()> {
     Ok(())
 }
 
+// Helper function to find git repository root by searching for .git directory
+fn find_git_root() -> Option<std::path::PathBuf> {
+    // Start from current directory
+    let mut current = std::env::current_dir().ok()?;
+
+    loop {
+        let git_dir = current.join(".git");
+        if git_dir.exists() {
+            return Some(current);
+        }
+
+        // Move up to parent directory
+        if !current.pop() {
+            // Reached filesystem root without finding .git
+            return None;
+        }
+    }
+}
+
 // Helper function to resolve defaults directory path
 // Tries development path first, then falls back to bundled resource path
 fn resolve_defaults_path(defaults_path: &str) -> Result<std::path::PathBuf, String> {
     use std::path::PathBuf;
 
-    // Try the provided path first (development mode)
+    // Try the provided path first (development mode - relative to cwd)
     let dev_path = PathBuf::from(defaults_path);
     if dev_path.exists() {
         return Ok(dev_path);
+    }
+
+    // Try finding defaults relative to git repository root
+    // This handles the case where we're running from a git worktree
+    if let Some(git_root) = find_git_root() {
+        let git_root_defaults = git_root.join(defaults_path);
+        if git_root_defaults.exists() {
+            return Ok(git_root_defaults);
+        }
     }
 
     // Try resolving as bundled resource (production mode)
@@ -339,7 +367,7 @@ fn resolve_defaults_path(defaults_path: &str) -> Result<std::path::PathBuf, Stri
     }
 
     Err(format!(
-        "Defaults directory not found: tried {defaults_path} and bundled resources"
+        "Defaults directory not found: tried {defaults_path}, git root, and bundled resources"
     ))
 }
 

